datatypes/src/operations/reproject.rs

```diff
@@ -214,14 +226,60 @@ where
             .chain(left_line)
             .collect();
 
+        let proj_outline_coordinates = projector.project_coordinates(&outline_coordinates)?;
+
+        Ok(MultiPoint::new_unchecked(proj_outline_coordinates).spatial_bounds())
+    }
+}
+
+pub trait ReprojectClipped<P: CoordinateProjection> {
+    type Out;
+    /// Reproject and clip with respect to the area of use of the projection
+    fn reproject_clipped(&self, projector: &P) -> Result<Self::Out>;
+}
+
+impl<P> ReprojectClipped<P> for BoundingBox2D
+where
+    P: CoordinateProjection,
+{
+    type Out = BoundingBox2D;
+    fn reproject_clipped(&self, projector: &P) -> Result<BoundingBox2D> {
+        const POINTS_PER_LINE: i32 = 7;
+
+        let use_bbox = projector.target_srs().area_of_use()?;
```

Comment:
i guess we need to transform the usebox to the source srs before using it? Its lat/lon right?

Reply:
7c3dcc44c62f4ead9564512f5d7b4f05479b4d43